node/tests/SharedTests.ts

```diff
@@ -873,6 +873,58 @@ export function runBaseTests<Context>(config: {
         config.timeout,
     );
 
+    it.each([ProtocolVersion.RESP2, ProtocolVersion.RESP3])(
+        `lset test_%p`,
+        async (protocol) => {
+            await runTest(async (client: BaseClient) => {
+                const key = uuidv4();
+                const nonExistingKey = uuidv4();
+                const index = 0;
+                const oobIndex = 10;
+                const negativeIndex = -1;
+                const element = "zero";
+                const lpushArgs = ["four", "three", "two", "one"];
+                const expectedList = ["zero", "two", "three", "four"];
+                const expectedList2 = ["zero", "two", "three", "zero"];
+
+                // key does not exist
+                await expect(
+                    client.lset(nonExistingKey, index, element),
+                ).rejects.toThrow();
```

Comment:
Can you add this arg to "toThrow" so that it checks the error type? same for the other "toThrow" calls in this test
```suggestion
                ).rejects.toThrow(RequestError);
```